docs/tutorials/basics/70_blocks.md

```diff
@@ -0,0 +1,570 @@
+end
+```
+
+### Unpacking parameters
+
+Now let's suppose we have an array of arrays. And we want to print each  array in the following way: *the first element followed by a hyphen, followed by the second element*.
+
+One way to implement this would be:
+
+```crystal-play
+arr = [["one", 42], ["two", 24]]
+arr.each do |arg|
+  puts "#{arg[0]} - #{arg[1]}"
+end
+```
+
+This works, but there is a more readable way of writing the same but using parameter unpacking::
+
+```crystal-play
+arr = [["one", 42], ["two", 24]]
+arr.each do |(word, number)|
+  puts "#{word} - #{number}"
+end
+```
+
+NOTE:
+We use parentheses to unpack the argument into the different block parameters.
+
+Unpacking arguments into parameters works only if the argument's type responds to `[i]` (with `i` an `integer`). In our example `Array` inherits [Indexable#[ ]](https://crystal-lang.org/api/Indexable.html#%5B%5D%28index%3AInt%29-instance-method)
+
+### Splats
+
+When the *block* parameter is a [Tuple](../../syntax_and_semantics/literals/tuple.md) we can use auto-splatting (see [Splats](../../syntax_and_semantics/operators.md#splats)) as a way of destructuring the `tuple` in block parameters (and without the need of parentheses).
+
+```crystal-play
+arr = [{"one", 42}, {"two", 24}]
+arr.each do |word, number|
+  puts "#{word} - #{number}"
+end
+```
+
+NOTE:
+`Tuples` also implements [Tuple#[ ]](https://crystal-lang.org/api/Tuple.html#%5B%5D%28index%3AInt%29-instance-method) meaning that we can also use *unpacking*.
+
+## Blocks' returned value
+
+A *block*, by default, returns the value of the last expression (the same as a method).
+
+```crystal-play
+def with_number(n : Int32)
+  block_result = yield n
+  puts block_result
+end
+
+with_number(41) do |number|
+  number + 1
+end
+```
+
+### Returning keywords
+
+We can use the `return` keyword ... but, let's see the following example:
+
+```crystal-play
+def with_number(n : Int32)
+  block_result = yield n
+  puts block_result
+end
+
+def test_number(n)
+  with_number(n) do |number|
+    return number if number.negative?
+    number + 1
+  end
+
+  puts "Inside `#test_number` method after `#with_number`"
+end
+
+test_number(42)
+```
+
+Outputs:
+
+```console
+
+43
+Inside `#test_number` method after `#with_number`
+```
+
+And if we want to `test_number(-1)` we would expect:
+
+```console
+
+-1
+Inside `#test_number` method after `#with_number`
+```
+
+Let's see ...
+
+```crystal-play
+def with_number(n : Int32)
+  block_result = yield n
+  puts block_result
+end
+
+def test_number(n)
+  with_number(n) do |number|
+    return number if number.negative?
+    number + 1
+  end
+
+  puts "Inside `#test_number` method after `#with_number`"
+end
+
+test_number(-1)
+```
+
+The output is empty! This is because Crystal implements *full closures*, meaning that using `return` inside the block will return, not only from the *block* itself but, from the method where the *block* is defined (i.e. `#test_number` in the above example).
```

Comment:
`test_number` is not an instance method (but a top-level one).

```suggestion
The output is empty! This is because Crystal implements *full closures*, meaning that using `return` inside the block will return, not only from the block itself but, from the method where the block is defined (i.e. `test_number` in the above example).
```